### ISSUE TITLE
Feature Request: Add ARN to aws_waf_ipset

### ISSUE THREAD

**Issue by BookOfGreg** [terraform-providers/terraform-provider-aws#4772]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Currently you need to construct your own ARNs by hand for `aws_waf_ipset`.
It would be helpful when crafting policies for this resource type.

### New or Affected Resource(s)

* aws_waf_ipset

### Potential Terraform Configuration

```hcl
resource "aws_iam_policy" "waf_update_policy" {
  name        = "WAFUpdatePolicy"
  description = "A WAF policy"
  policy = <<EOF
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Action": [
        "waf:GetIPSet",
        "waf:UpdateIPSet"
      ],
      "Effect": "Allow",
      "Resource": [
        "${aws_waf_ipset.ip_rep_1_ipset.arn}",
        "${aws_waf_ipset.ip_rep_2_ipset.arn}"
      ]
    },
    {
      "Action": "waf:GetChangeToken",
      "Resource": "*",
      "Effect": "Allow"
    }
  ]
}
EOF
}
```

Currently you have to do this:
```
data "aws_caller_identity" "current" {}
resource "aws_iam_policy" "waf_update_policy" {
  name        = "WAFUpdatePolicy"
  description = "A test policy"
  policy = <<EOF
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Action": [
        "waf:GetIPSet",
        "waf:UpdateIPSet"
      ],
      "Effect": "Allow",
      "Resource": [
        "arn:aws:waf::${data.aws_caller_identity.current.account_id}:ipset/${aws_waf_ipset.ip_rep_1_ipset.id}",
        "arn:aws:waf::${data.aws_caller_identity.current.account_id}:ipset/${aws_waf_ipset.ip_rep_2_ipset.id}"
      ]
    },
    {
      "Action": "waf:GetChangeToken",
      "Resource": "*",
      "Effect": "Allow"
    }
  ]
}
EOF
}
```

**Comment by bflad** [comment]
The `arn` attribute for the `aws_waf_ipset` resource will be available in version 1.23.0 of the AWS provider, likely tomorrow.

**Comment by bflad** [comment]
This has been released in [version 1.23.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1230-june-14-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
